File: src/main/java/dev/tricked/hardermc/tools/CustomRecipe.java
package dev.tricked.hardermc.tools;

import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.inventory.Recipe;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;
import java.util.List;

abstract public class CustomRecipe {
    public @NotNull NamespacedKey key;
    public @NotNull Recipe recipe;
    public @NotNull List<Material> triggerIngredients;

    public void init(@NotNull NamespacedKey key, @NotNull Recipe recipe, @NotNull Material[] triggerIngredients) {
        this.key = key;
        this.recipe = recipe;
        this.triggerIngredients = Arrays.stream(triggerIngredients).toList();
    }

}

File: src/main/kotlin/dev/tricked/hardermc/utilities/CustomRecipe.kt
/*
 * Copyright (c) Tricked-dev 2024.
 */

package dev.tricked.hardermc.utilities

import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.inventory.Recipe

abstract class CustomRecipe {
    var key: NamespacedKey? = null
    var recipe: Recipe? = null
    var triggerIngredients: List<Material> = ArrayList()
    fun init(key: NamespacedKey, recipe: Recipe, triggerIngredients: Array<Material>) {
        this.key = key
        this.recipe = recipe
        this.triggerIngredients = triggerIngredients.toList()
    }

    fun setRecipeKey(key: NamespacedKey) {
        this.key = key
    }
}
